packages/contracts/contracts/registry/DAORegistry.sol

```diff
@@ -3,40 +3,53 @@
 
 import "../core/IDAO.sol";
+import "./ens/ENSSubdomainRegistrar.sol";
 import "./InterfaceBasedRegistry.sol";
 
 /// @title Register your unique DAO name
 /// @author Aragon Association - 2022
 /// @notice This contract provides the possiblity to register a DAO.
 contract DAORegistry is InterfaceBasedRegistry {
+    /// @notice The ID of the permission required to call the `register` function.
     bytes32 public constant REGISTER_DAO_PERMISSION_ID = keccak256("REGISTER_DAO_PERMISSION");
 
-    /// @notice Emitted when a new DAO is registered
-    /// @param dao The address of the DAO contract
-    /// @param creator The address of the creator
-    /// @param name The name of the DAO
-    event DAORegistered(IDAO indexed dao, address indexed creator, string name);
-
-    /// @notice Initializes the contract
-    /// @param _managingDao the managing DAO address
-    function initialize(IDAO _managingDao) public initializer {
+    /// @notice The ENS subdomain registrar registering the DAO names.
+    ENSSubdomainRegistrar private subdomainRegistrar;
+
+    /// @notice Emitted when a new DAO is registered.
+    /// @param dao The address of the DAO contract.
+    /// @param creator The address of the creator.
+    /// @param name The DAO name.
+    event DAORegistered(address indexed dao, address indexed creator, string name);
+
+    /// @notice Initializes the contract.
+    /// @param _managingDao the managing DAO address.
+    function initialize(IDAO _managingDao, ENSSubdomainRegistrar _subdomainRegistrar)
+        public
+        initializer
+    {
         __InterfaceBasedRegistry_init(_managingDao, type(IDAO).interfaceId);
+        subdomainRegistrar = _subdomainRegistrar;
     }
 
-    /// @notice Registers a DAO by its address
-    /// @dev A name is unique within the Aragon DAO framework and can get stored here
-    /// @param name The name of the DAO
-    /// @param dao The address of the DAO contract
-    /// @param creator The address of the creator
+    /// @notice Registers a DAO by its address.
+    /// @dev A name is unique within the Aragon DAO framework and can get stored here.
+    /// @param _dao The address of the DAO contract.
+    /// @param _creator The address of the creator.
+    /// @param _name The DAO name.
     function register(
-        string calldata name,
-        IDAO dao,
-        address creator
+        IDAO _dao,
+        address _creator,
+        string calldata _name
     ) external auth(REGISTER_DAO_PERMISSION_ID) {
-        // TODO: Implement ENS subdomain. Currently DAO's name can be repeated, will be resolved once the ENS subdomain is implemented.
+        address daoAddr = address(_dao);
+
+        _register(daoAddr);
+
+        bytes32 labelhash = keccak256(bytes(_name));
 
-        _register(address(dao));
+        subdomainRegistrar.registerSubnode(labelhash, daoAddr);
 
-        emit DAORegistered(dao, creator, name);
+        emit DAORegistered(daoAddr, _creator, _name);
```

Comment:
there is no need for the node to be declared in the DAORegistered(...) event?

Reply:
Currently, it is not used anywhere. If needed, we can add it.